pkg/generators/snippets/setters.go

```diff
@@ -0,0 +1,83 @@
+package snippets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/gengo/generator"
+	"k8s.io/gengo/types"
+)
+
+func GenerateSetter(w io.Writer, ctx *generator.Context, root *types.Type, member types.Member, hasParent bool, embedded *types.Type) error {
```

Comment:
Need a second opinion to see if this function argument list is okay. The arguments of concern are:
`root`, the root struct that setter is acting on
`member`, the field that the setter needs to set
`hasParent`, does the `member` belong to an intermediate embedded type of `root`
`embedded`. is the `member` a struct that embeds a type? If so, specify the embedded type here.

For example:
```
// +kanopy:builder=true
type Namespace struct {
	corev1.Namespace
	jsonpatch.StructField
}

// +kanopy:builder=true
type NamespaceSpec struct {
	corev1.NamespaceSpec
}

// In k8s.io/api/core/v1 types.go
type Namespace struct {
    metav1.TypeMeta
    metav1.ObjectMeta
    Spec NamespaceSpec
}

type ObjectMeta struct {
    Name string
}
```

To set Namespace.ObjectMeta.Name, call GenerateSetter with
- `root`, our top level Namespace struct
- `member`, Name member
- `hasParent`, `true` because Name belongs to the embedded Namespace type
-  `embedded`, nil because Name itself is not a complex type that embeds another type

To set Namespace.NamespaceSpec, call GenerateSetter with
- `root`, our top level Namespace struct
- `member`, Spec member
- `hasParent`, `true` because Name belongs to the embedded Namespace type
-  `embedded`, corev1.NamespaceSpec type because it is the complex type embedded within our wrapper NamespaceSpec

Reply:
I agree the argument list is a bit long.  I think we can clean these up by splitting the different setter functions.